fs/read.js

```diff
@@ -5,18 +5,98 @@ var fs = require('fs');
 
 // 1. Write script to read file theory.md and console the output buffer.
 
+  // fs.readFile('./theory.md', (err, data) => {
+  //   if(err) {
+  //     console.log('Error: ', err);
+  //   } else {
+  //       console.log(data)
+  //   }
+  // })
+
 // 2. Write script to read file theory.md and convert resulted buffer using toString
   // method and console the result.
 
+  // fs.readFile('./theory.md', (err, data) => {
+  //   if(err) {
+  //     console.log('Error: ', err);
+  //   } else {
+  //       console.log(data.toString());
+  //   }
+  // })
+
 // 3. Write script to read file Synchronously and console the output.
 
+  // var data = fs.readFileSync('./theory.md');
+  // console.log(data);
+
 // 4. Write script to create a file 'write.js' and write content of read.js in there.
 
+  // fs.readFile(__filename, (err, data) => {
+  //   if(err) {
+  //     console.log('Error: ', err);
+  //   } else {
+  //       fs.writeFile('write.js', data.toString(), (err) => {
+  //         if(err) {
+  //           console.log('Error: ', err);
+  //         } else {
+  //             console.log('Data Saved!');
+  //             return;
+  //         }
+  //       })
+  //   }
+  // })
+
 // 5. Write script to update content of write.js. Update it with content of theory.md
 //   Steps are
 //   - open the file(write.js) using fs.open
 //   - remove earlier content using fs.ftruncate
 //   - add new content using fs.writeFile
 //   - close the file at last using fs.close
 
-// 6. Delete the content of write.js using fs.unlink or unlinkSync method
\ No newline at end of file
+  // fs.open('./write.js',(err, fd) => {
```

Comment:
Instead of using if/else conditional, a better approach would be to return in case of error.
```js
fs.open('./write.js',(err, fd) => {
  if(err) return console.log('Error: ', err);
  fs.truncate('./write.js',fd.length, (err) => {
    if(err) return console.log('Error: ', err);
    // And so on....
})
})
```